src/types/decimal.rs

```diff
@@ -311,7 +311,18 @@ impl TryFrom<f64> for Decimal {
         let coefficient = value * 10f64.powi(PRECISION as i32);
         let exponent = -(PRECISION as i64);
 
-        Ok(Decimal::new(coefficient as i64, exponent))
+        // prefer a compact representation for the coefficient; fallback to bigint
+        let coefficient: Coefficient = if coefficient <= i64::MAX as f64 {
+            (coefficient as i64).into()
+        } else {
+            coefficient
+                .trunc()
+                .to_bigint()
+                .ok_or_else(|| illegal_operation_raw("Cannot convert large f64 to Decimal."))?
```

Comment:
Could you add a unit test that exercises this code path? Glancing at [`num_bigint`'s implementation of `BigInt::from_f64`](https://docs.rs/num-traits/0.2.15/src/num_traits/cast.rs.html#516-522), it will succeed for values up to `u64::MAX` (not `i64::MAX`).

Reply:
I think that's the wrong crate.

The `to_bigint` runs through [this](https://docs.rs/num-bigint/latest/src/num_bigint/bigint/convert.rs.html#170-177), which in turn runs through [this](https://docs.rs/num-bigint/latest/src/num_bigint/biguint/convert.rs.html#444-471)